projects/plugins/jetpack/extensions/blocks/paywall/editor.js

```diff
@@ -1,4 +1,20 @@
+import { unregisterBlockType } from '@wordpress/blocks';
+import { subscribe, select } from '@wordpress/data';
 import registerJetpackBlock from '../../shared/register-jetpack-block';
 import { name, settings } from '.';
 
+const unsubscribe = subscribe( () => {
+	const postType = select( 'core/editor' ).getCurrentPostType();
+
+	// If postType is still not available, simply return and wait for the next call.
+	if ( postType === null ) {
+		return;
+	}
+	unsubscribe();
+	// If postType is defined and not 'post', unregister the block.
+	if ( postType && postType !== 'post' ) {
+		unregisterBlockType( 'jetpack/' + name );
+	}
+} );
+
```

Comment:
Looks like we using recursion here. I was wondering if you could explain the logic? I think subscribe adds a listener that will run any time state changes. Since it already runs each time state changes, I'm not understanding why we'd need the recursive call.

Reply:
it's not recursive, it calls `unsubscribe`, so we don't try to unregister multiple times!

Reply:
So from separate conversation, the WP subscribe() method returns unsubscribe, which can be called later to undo the subscribe() call (ie, remove the listener so it doesn't keep re-running each time state is updated). 

We call it here once we have a post type, since we do not need to continue running this code after that point. 

Seems like a bit of an odd construct to have unsubcribe() returned from subscribe() and also called within subscribe(), but also sounds like this is not an uncommon pattern with this particular method.

Reply:
🙌 thanks for the 🕵️ work